communitiesuk/funding-service-design-assessment
# FS-3235: Generate PDF representation of application

### Change description

- Refactor file generation functionality into own module and add tests
- Add PDF generation as an option for application exports
- Change round/fund models to include changes from fund-store
- Add [pdf_generator.py](https://github.com/communitiesuk/funding-service-design-assessment/compare/main...tferns-fs-3235-pdf-first-pass#diff-cfe1a5fedb5965dbfd9ab52fa77170215956199b12cb18485a428a6926093907) for generic PDF generation functionality
- Add [full_application.py](https://github.com/communitiesuk/funding-service-design-assessment/compare/main...tferns-fs-3235-pdf-first-pass#diff-0f0ea0fe2b8b0029fa7a94201f0ddea76ec8547b7653c7cefd3cfc3acd5d4a10) for an example of a concrete PDF generation template
- Other test coverage & changes
- JavaScript if statement just to prevent errors on other pages
- Text & CSV files should now also be ordered properly by application order

---


### How to test

- Make sure you have this branch & the [fund-store](https://github.com/communitiesuk/funding-service-design-fund-store/pull/113) PR checked out
- Go to any application you can find on assessment
- Add `/export` to the url i.e `/assess/application/application_id/export`
- Right click the application answers url, and copy to clipboard
    - `/assess/application/b98d21c3-f1ac-4125-a01a-02156a12bbdf/export/TCZSYK/answers.txt`
- Change the end of the url to `.pdf` to download the PDF, open and observe PDF

### Example

[TCZSYK_answers_no_logo.pdf](https://github.com/communitiesuk/funding-service-design-assessment/files/12312230/TCZSYK_answers_no_logo.pdf)
